### PR TITLE
Order service methods 

### PR DESCRIPTION
**Pull request by Islom-35**

### CODE REVIEW

File: internal/order/adapters/order_postgres_repo.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		"INSERT INTO orders (number, basket_id, total_price, status, created_at, updated_at) VALUES ($1, $2, $3, $4, $5, $6) RETURNING id",
+		order.Number, order.BasketID, order.TotalPrice, order.Status, order.CreatedAt, order.UpdatedAt).
+		Scan(&order.ID)
+
+	return err
+}
+
+func (o *orderRepo) GetOrderByID(orderID int) (domain.Order, error) {
+	var order domain.Order
+	fmt.Println("order id in repo: ", orderID)
```
Comment (khudayberdiyev25) [inline internal/order/adapters/order_postgres_repo.go:29]: why we need this print statement ?

File: internal/order/app/order_service.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+func NewOrderService(repo domain.OrderRepository, basketservice BasketService, productservice app.ProductService) OrderService {
+	return &orderService{
+		repo:           repo,
+		basketService:  basketservice,
+		productService: productservice,
+	}
+}
+
+func (o *orderService) CreateOrder(basketID int) error {
+	basketItems, err := o.basketService.GetAll(basketID)
```
Comment (khudayberdiyev25) [inline internal/order/app/order_service.go:33]: basketService.GetItems ?

File: internal/order/app/order_service.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+	if err != nil {
+		return err
+	}
+
+	var totalAmount int
+
+	for _, item := range basketItems {
+
+		product, err := o.productService.GetOne(item.ProductId)
```
Comment (khudayberdiyev25) [inline internal/order/app/order_service.go:42]: don't we have price and quantity in BasketItem struct ?

File: internal/order/app/order_service.go [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+func NewOrderService(repo domain.OrderRepository, basketservice BasketService, productservice app.ProductService) OrderService {
+	return &orderService{
+		repo:           repo,
+		basketService:  basketservice,
+		productService: productservice,
+	}
+}
+
+func (o *orderService) CreateOrder(basketID int) error {
```
Comment (khudayberdiyev25) [inline internal/order/app/order_service.go:32]: How you know which client is purchasing order ?

**Review by khudayberdiyev25 — APPROVED** [review]

File: internal/order/adapters/order_postgres_repo.go (hunk 1 above)
Comment (Islom-35) [reply internal/order/adapters/order_postgres_repo.go:29]: for my self I wrote this printer

File: internal/order/app/order_service.go (hunk 2 above)
Comment (Islom-35) [reply internal/order/app/order_service.go:33]: yes that method gives us basket Items and I used in for

File: internal/order/app/order_service.go (hunk 4 above)
Comment (Islom-35) [reply internal/order/app/order_service.go:32]: yes,sorry I forgot use that method

Comment (Islom-35) [reply internal/order/app/order_service.go]: I added that method

**Pull request merged**